src/Templates/AbstractTemplateCompatibility.php

```diff
@@ -0,0 +1,179 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+/**
+ * BlockTemplatesCompatibility class.
+ *
+ * To bridge the gap on compatibility with PHP hooks and blockified templates.
+ *
+ * @internal
+ */
+abstract class AbstractTemplateCompatibility {
+	/**
+	 * The data of supported hooks, containing the hook name, the block name,
+	 * position, and the callbacks.
+	 *
+	 * @var array $hook_data The hook data.
+	 */
+	protected $hook_data;
+
+	/**
+	 * Initialization method.
+	 */
+	public function init() {
+		if ( ! wc_current_theme_is_fse_theme() ) {
+			return;
+		}
+
+		$this->set_hook_data();
+
+		add_filter( 'render_block_data', array( $this, 'update_render_block_data' ), 10, 3 );
+		add_filter(
+			'render_block',
+			function ( $block_content, $block ) {
+				/**
+				* Filters to disable the compatibility layer for the blockified templates.
+				*
+				* This hooks allows to disable the compatibility layer for the blockified.
+				*
+				* @since TBD
+				* @param boolean.
+				*/
+				$is_disabled_compatility_layer = apply_filters( 'woocommerce_disable_compatibility_layer', false );
+
+				if ( $is_disabled_compatility_layer ) {
+					return $block_content;
+				}
+
+				$this->inject_hooks( $block_content, $block );
```

Comment:
Should this value be returned as well?
```suggestion
				return $this->inject_hooks( $block_content, $block );
```